nationalarchives/ds-ohos-wagtail
# HERITAGE-149:results, details pages level access updates

Ticket URL: https://national-archives.atlassian.net/browse/HERITAGE-149

## About these changes

- Integrates API changes for level with the search results and details pages
- Updates Collection label for Filter, Search results and Details pages
- Adds Community archive web page url in details page
- Refactors Collection names 
- Config attributes 
   - `CommunityCollectionMapping` : Collection/Archive Names, Community Level ciim id, webpage urls
   - `CommunityLevels` - levels for community record
 - Add, Update Tests
 - Add default "entries" from API response form empy aggs - which would otherwise error out (see example)

## How to check these changes

- Go to search results - The label which was Collection will also Community Archive
- Go to detail record - There would be a row for Collection, Community Archive or both depending on the collection and level of the record  - see ticket.
- On details page click on Collection link - should navigate within OHOS on same tab.
- On details page click on Community Archive link - should navigate to community web page on new tab.

http://127.0.0.1:8000/catalogue/id/shc-9967-1-18/  (Item->Series->Collection->Community)
http://127.0.0.1:8000/catalogue/id/swop-2897/ (Item->Community)
http://127.0.0.1:8000/catalogue/id/mpa-13090/ (Item->Collection->Community)
http://127.0.0.1:8000/catalogue/id/wmk-20479/ (Item->Collection)
http://127.0.0.1:8000/catalogue/id/pcw-383261/ (Item->Community)

Default "entries" for empty Agg example
http://127.0.0.1:8000/search/catalogue/?q=MK50&&collection=parent-collectionWMK%3AMilton+Keynes+City+Discovery+Centre&vis_view=list&group=community

## Before assigning to reviewer, please make sure you have


## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)
